tensorflow/lite/experimental/micro/examples/micro_speech/arduino-esp32/audio_provider.cc

```diff
@@ -0,0 +1,204 @@
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/lite/experimental/micro/examples/micro_speech/audio_provider.h"
+
+#include <Arduino.h>        // NOLINT
+#include <driver/i2s.h>
+#include <soc/i2s_reg.h> // I2S_TIMING_REG
+
+#include "tensorflow/lite/experimental/micro/examples/micro_speech/micro_features/micro_model_settings.h"
+
+namespace {
+static const i2s_port_t i2s_num = I2S_NUM_0; // i2s port number
+
+// These are the raw buffers that are filled by the ADC during DMA
+constexpr int kAdcNumSlots = 8;
+constexpr int kAdcSamplesPerSlot = 128;
+
+TaskHandle_t xCaptureTaskHandle = NULL;
+
+bool g_is_audio_initialized = false;
+constexpr int kAudioCaptureBufferSize = kAudioSampleFrequency * 0.5;
```

Comment:
why did you use  `const` in `constants.cc` but `constexpr` here?

Reply:
kAudioCaptureBufferSize has to be known at compile-time if the buffer is allocated statically. This is the case in all the other examples - except this one - but i wanted to stick close to the 'recommended' static allocation. (and not use #define)